### PR TITLE
call stop functions in reverse order

### PR DESCRIPTION
**Pull request by DavidVorick**

### CODE REVIEW

File: sync/threadgroup.go [hunk 1]
```diff
@@ -84,7 +85,7 @@ func (tg *ThreadGroup) Stop() error {
 }
 
 // OnStop adds a function to the ThreadGroup's stopFns set. Members of the set
-// will be closed when Stop is called.
+// will be closed when Stop is called, in reverse order.
```
Comment (lukechampine) [inline sync/threadgroup.go]: should be `will be called`, forgot to fix it last time

**Comment by lukechampine** [comment]
lol I just made this exact patch.

File: sync/threadgroup.go [hunk 2]
```diff
@@ -75,7 +75,8 @@ func (tg *ThreadGroup) Stop() error {
 	}
 	close(tg.stopChan)
 	for _, fn := range tg.stopFns {
-		fn()
+	for i := len(tg.stopFns)-1; i >= 0; i-- {
+		tg.stopFns[i]()
```
Comment (lukechampine) [inline sync/threadgroup.go]: took me way too long to figure out what was wrong with this diff...

Comment (DavidVorick) [inline sync/threadgroup.go]: darn

**Comment by DavidVorick** [comment]
fixed

**Pull request merged**